datasets/li/posting_sanctions/crawler.py

```diff
@@ -77,88 +75,98 @@ def parse_target(
-    company.add("name", company_name)
-    h.apply_address(context, company, address)
-
     emp = context.make("Employment")
     emp.id = context.make_id("Employment", person.id, company.id)
     emp.add("employee", person)
     emp.add("employer", company)
-    h.apply_date(emp, "date", date)
+    emp.add("date", date)
     emp.add("role", "Manager found responsible for breaking the law")
     context.emit(person, target=False)
     context.emit(emp, target=False)
 
     return company
 
 
-def parse_debarments(context: Context, doc) -> None:
-    table = doc.xpath(
-        "//h2[text()='Laufende und abgelaufene Entsendesperren"
-        + " (Art. 7 Abs. 2 Entsendegesetz)']/following::table[1]"
-    )[0]
-    for row in table.xpath("tbody/tr")[1:]:
-        [start, name, addr, law, end] = row.xpath("descendant::*/text()")
-        address = parse_address(context, addr)
+def extract_rows(path: Path) -> List[Dict[str, str]]:
+    pdf = pdfplumber.open(path.as_posix())
+    settings = {
+        "vertical_strategy": "lines",
+        "horizontal_strategy": "lines",
+        "text_tolerance": 1,
+    }
+    headers: Optional[List[str]] = None
+    rows: List[Dict[str, str]] = []
+    for page in pdf.pages:
+        cropped = page.crop((0, 50, page.width, page.height - 10))
+        for row in cropped.extract_table(settings):
+            if headers is None:
+                headers = row
+                continue
+            rows.append({k: v for k, v in zip(headers, row)})
+    return rows
+
+
+def crawl_debarments(context: Context) -> None:
+    path = context.fetch_resource("sperren.pdf", DEBARMENT_URL)
+    for row in extract_rows(path):
+        if len(row) != 5:
+            continue
+        address = parse_address(context, row.pop("Adresse"))
+        name = row.pop("Betrieb")
+        start = h.parse_date(row.pop("In Rechtskraft"), ["%d.%m.%Y"])
+        end = h.parse_date(row.pop("Ende der Sperre"), ["%d.%m.%Y"])
         company = parse_target(context, name, address, start)
         if company is None:
             continue
         company.add("topics", "debarment")
+        violation = row.pop("Verstoss")
         sanction = h.make_sanction(context, company)
         sanction.id = context.make_id(
-            "Sanction", "Debarment", company.id, law, start, end
+            "Sanction", "Debarment", company.id, violation, start, end
         )
-        h.apply_date(sanction, "date", start)
-        h.apply_date(sanction, "endDate", end)
+        sanction.add("startDate", start)
+        sanction.add("endDate", end)
         sanction.add("description", "Debarment")
         sanction.add("program", "EntsG Sanctions")
         reason = (
             "Repeated or severe infraction against "
-            f"Liechtenstein Posted Workers Act, {law}"
+            f"Liechtenstein Posted Workers Act, {violation}"
         )
         sanction.add("reason", reason)
         context.emit(sanction)
         context.emit(company, target=True)
```

Comment:
we probably want some `ended = max(end) < settings.RUN_TIME_ISO` on topics and target, right?

I'll get to that helper any day now